### PR TITLE
Render layers as a group

### LINKED ISSUE #7133: Render layers as groups (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#46183
Author: @nyalldawson
QGIS version: 3.24

**Render layers as groups**

### PR Description:
This exposes the ability to render group layers, as described by qgis/QGIS-Enhancement-Proposals#235.

When a group is selected in the layer tree the styling dock will show a new panel for controlling the appearance of the group as a whole, with options for group opacity, blend mode and layer effects.

image

Sponsored by Andrew Fletcher.

### Commits tagged with [need-docs] or [FEATURE]
"[feature] When a group is selected in the layer tree, the layer\nstyling dock now shows an option to control the rendering\nappearance of the entire group\n\nBy selecting \"Render Layers as a Group\", the whole set of layers\npresent in the group will be rendered as a flattened object during\nmap renders. This means that the opacity and blend mode assigned\nto the group will apply to the flattened version of all the\nchild layers, instead of applying to each layer one-by-one.\n\nThis allows for styling results such as groups of layers which\nobscure each other, but overall are rendered as semi-opaque."

### PR DESCRIPTION
**Pull request by DelazJ**
Fixes #7133 
Supersedes #7155

### CODE REVIEW

File: docs/user_manual/introduction/general_tools.rst [hunk 1]
```diff
@@ -2791,6 +2845,34 @@ on layers and features, and also on print layout items:
...
+They provide methods to clip the render of one layer’s content by the content in a second "mask" layer.
+
+* :guilabel:`Masked By Below`: The output is the source, where the alpha is reduced by that of the destination.
+* :guilabel:`Mask Below`: The output is the destination, where the alpha is reduced by that of the source.
+* :guilabel:`Inverse Masked By Below`: The output is the source, where the alpha is reduced by the inverse of destination.
+* :guilabel:`Inverse Mask Below`: The output is the destination, where the alpha is reduced by the inverse of the source.
+* :guilabel:`Paint Inside Below`: The source pixel is blended on top of the destination,
+  with the alpha of the source pixel reduced by the alpha of the destination pixel.
+* :guilabel:`Paint Below Inside`: The destination pixel is blended on top of the source,
+  with the alpha of the destination pixel is reduced by the alpha of the destination pixel.
```
Comment (DelazJ) [inline docs/user_manual/introduction/general_tools.rst:2863]: TBH I picked this from the source code and I'm unsure of how understandable this is for end users. What do `source` and `destination` refer to? top and bottom layers?

File: docs/user_manual/introduction/general_tools.rst [hunk 2]
```diff
@@ -336,6 +335,57 @@ These elements are:
...
+.. _render_as_group:
+
+Control layers rendering through grouping
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Groups are a means of structuring layers within a tree in the project
+but they can also impact how their component layers are rendered,
+namely as a single flattened object during map renders.
+
+Option for such a rendering is available within the :guilabel:`Layer Styling` panel whenever a group is selected.
```
Comment (nyalldawson) [inline docs/user_manual/introduction/general_tools.rst:347]:
```suggestion
The option for such a rendering is available within the :guilabel:`Layer Styling` panel whenever a group is selected.
```

File: docs/user_manual/introduction/general_tools.rst [hunk 3]
```diff
@@ -336,6 +335,57 @@ These elements are:
...
+     and then the result is rendered at 50% opacity).
+
+* :guilabel:`Blend modes`: Just like opacity, setting a :ref:`blend mode <blend-modes>` (like multiply, overlay, ...)
+  for an entire group results first in flattening features of child layers, with upper ones obscuring lower.
+  The rendering is then obtained by blending the flat group and the layers sitting below the group.
+
+  * When the child layers have blend modes assigned, it is applied before flattening
+    but the scope is restricted to only affecting other child layers from that group,
+    and not other layers sitting below the whole group.
+  * Child layers are added some more :ref:`blending modes <blending_clipping>` options
```
Comment (nyalldawson) [inline docs/user_manual/introduction/general_tools.rst:374]:
```suggestion
  * Some more :ref:`blending modes <blending_clipping>` options are available for child layers in groups,
```

File: docs/user_manual/introduction/general_tools.rst (hunk 1 above)
Comment (nyalldawson) [reply docs/user_manual/introduction/general_tools.rst:2872]: Source is the top layer, destination is the bottom layer. You could probably just replace these words and the descriptions would be sufficient.

Comment (nyalldawson) [reply docs/user_manual/introduction/general_tools.rst:2872]: Looking good! I'd also change "alpha" to "opacity" too to make it a bit friendlier.

**Pull request merged**